plot heatmap tool

dzesikahoinkis opened the pull request.

dzesikahoinkis committed:
plot heatmap tool

nsoranzo commented on plotheatmap/.shed.yml:
```diff
@@ -0,0 +1,4 @@
+categories: [Computational chemistry]
+description: Additional tools for EDGE-PRO
+name: rnaseq_pro_workflow_tools
+owner: marpiech
```
owner should be `earlhaminst`

nsoranzo commented on plotheatmap/.shed.yml:
```diff
@@ -0,0 +1,4 @@
+categories: [Computational chemistry]
+description: Additional tools for EDGE-PRO
+name: rnaseq_pro_workflow_tools
```
s/rnaseq_pro_workflow_tools/plotheatmap/

nsoranzo commented on plotheatmap/.shed.yml:
```diff
@@ -0,0 +1,4 @@
+categories: [Computational chemistry]
+description: Additional tools for EDGE-PRO
```
Can you change this to a description of the `plotheatmap` tool?

nsoranzo commented:
@dzesikahoinkis Thanks for the contribution and very sorry for the late reply! Can you move the `plotheatmap` directory inside `tools/`, please?

nsoranzo commented on plotheatmap/tool_test_output.json:
```diff
@@ -0,0 +1 @@
+{"tests": [{"data": {"status": "success", "inputs": {"main|stats_matrix": {"src": "hda", "id": "5729865256bc2525"}, "filter|pvalue": "0.05", "filter|fdr": "0.1", "filter|log_fc": "1", "filter|log_cpm": "2", "main|count_matrix": {"src": "hda", "id": "2891970512fa2d5a"}}, "job": {"inputs": {"stats_matrix": {"src": "hda", "id": "5729865256bc2525", "uuid": "6f35fa40-8bb3-413f-b009-042c2a813b94"}, "co ...
```
You can safely remove this file.

nsoranzo commented on plotheatmap/tool_test_output.html:
```diff
@@ -0,0 +1,383 @@
+<!DOCTYPE html>
```
You can safely remove this file.

nsoranzo commented on plotheatmap/tools/script.R:
```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env Rscript
```
It'd be better to move this file inside `tools/plotheatmap/` .

nsoranzo commented on plotheatmap/plotheatmap.xml:
```diff
@@ -0,0 +1,103 @@
+        ]]>
+    </command>
+    <inputs>
+        <section name="main" title="Main parameters" expanded="True">
```
You can probably remove the `main` section opening and closing tags, and have the 2 data inputs always shown.

nsoranzo commented on plotheatmap/plotheatmap.xml:
```diff
@@ -0,0 +1,103 @@
+            <param name="log_cpm" type="float" value="0.5" label="A log CPM threshold"/>
+            <param name="pvalue" type="float" value="0.5" label="A PValue threshold">
+                <validator type="in_range" min="0" max="1" />
+            </param>
```
```xml
<param name="pvalue" type="float" min="0" max="1" value="0.5" label="A p-value threshold" />
```

nsoranzo commented on plotheatmap/plotheatmap.xml:
```diff
@@ -0,0 +1,103 @@
+            </param>
+            <param name="fdr" type="float" value="0.5" label="A FDR threshold">
+                <validator type="in_range" min="0" max="1" />            
+            </param>
```
```xml
<param name="fdr" type="float" min="0" max="1" value="0.5" label="An FDR threshold" />
```

nsoranzo commented on plotheatmap/plotheatmap.xml:
```diff
@@ -0,0 +1,103 @@
+            <param name="cluster_row" type="select" label="Determines if the row dendrogram should be reordered">
+                <option value="No" >No </option>
+                <option value="Yes" selected="true" >Yes </option>
+            </param>
```
```xml
<param name="cluster_row" type="boolean" truevalue="Yes" falsevalue="No" checked="true" label="Determines if the row dendrogram should be reordered" />
```

nsoranzo commented on plotheatmap/plotheatmap.xml:
```diff
@@ -0,0 +1,103 @@
+    <stdio>
+        <exit_code range="1:" />
+    </stdio>
+    <command>
```
You can remove the `<stdio>` section above and use:
```xml
<command detect_errors="aggressive">
```
which will also look for the 'error:' string in stderr, see https://docs.galaxyproject.org/en/latest/dev/schema.html#detect-errors

nsoranzo commented on plotheatmap/plotheatmap.xml:
```diff
@@ -0,0 +1,103 @@
+    </stdio>
+    <command>
+        <![CDATA[ 
+            $__tool_directory__/tools/script.R $main.count_matrix $main.stats_matrix $filter.log_fc $filter.log_cpm $filter.pvalue $filter.fdr $heatmap.cluster_row $heatmap.cluster_col $heatmap.hclust_method $plotparm.col_marg $plotparm.row_marg $plotparm.pdf_width $plotparm.pdf_height > $rscript_log 2> $rerror_log
```
```
R '$__tool_directory__/script.R'
'$count_matrix'
'$stats_matrix'
$filter.log_fc
$filter.log_cpm
$filter.pvalue
$filter.fdr
$heatmap.cluster_row
$heatmap.cluster_col
$heatmap.hclust_method
$plotparm.col_marg
$plotparm.row_marg
$plotparm.pdf_width
$plotparm.pdf_height
```

nsoranzo commented on plotheatmap/plotheatmap.xml:
```diff
@@ -0,0 +1,103 @@
+            <filter>(outftype == 'pdf')</filter>
+        </data>
+        <data name="rscript_log" format="data" />
+        <data name="rerror_log" format="data" />
```
I think you can remove the `rscript_log` and `rerror_log` outputs, they seems to unnecessarily clutter the history. These information will be available anyway when clicking the "i" icon of the `haetamappdf` output dataset.

nsoranzo commented on plotheatmap/plotheatmap.xml:
```diff
@@ -0,0 +1,103 @@
+            <param name="cluster_col" type="select" label="Determines if the column dendrogram should be reordered">
+                <option value="No" selected="true">No </option>
+                <option value="Yes" >Yes </option>
+            </param>
```
```xml
<param name="cluster_col" type="boolean" truevalue="Yes" falsevalue="No" label="Determines if the column dendrogram should be reordered" />
```

dzesikahoinkis committed:
Changes in plotheatmap tool were added

plotheatmap directory was moved to tools directory; changes in plotheatmap/plotheatmap.xml were added; changes in plotheatmap/.shed.yml were added; useless files were removed

nsoranzo commented on tools/plotheatmap/plotheatmap.xml:
```diff
@@ -0,0 +1,89 @@
+    </inputs>
+    <outputs>
+       <data format="pdf" name="haetamappdf" from_work_dir="heatmap.pdf">
+            <filter>(outftype == 'pdf')</filter>
```
`outftype` is not defined, so I guess you can remove the filter.

nsoranzo commented on tools/plotheatmap/plotheatmap.xml:
```diff
@@ -0,0 +1,89 @@
+        <![CDATA[ 
+Rscript $__tool_directory__/script.R 
+$count_matrix 
+$stats_matrix 
```
Can you add single quotes `'` around `$__tool_directory__/script.R`, `$count_matrix` and `$stats_matrix` please? So that spaces in filenames don't break the tool.

nsoranzo commented on tools/plotheatmap/.shed.yml:
```diff
@@ -0,0 +1,4 @@
+categories: [Computational chemistry]
+description: This tool can be used to plot heatmap of gene expression data. The genes are chosen based on p-value, FDR, log FC and log CPM from edgeR output. 
+name: rnaseq_pro_workflow_tools/plotheatmap/
```
Sorry, I didn't make myself clear, the name should be just `plotheatmap`.

nsoranzo reviewed (commented):
Thanks for the fixes @dzesikahoinkis!

dzesikahoinkis committed:
Update plotheatmap.xml and .shed.yml

nsoranzo commented:
`tools/plotheatmap/RnaSeqInProcaryotes.png` doesn't seem to be used.

nsoranzo commented on tools/plotheatmap/.shed.yml:
```diff
@@ -0,0 +1,4 @@
+categories: [Computational chemistry]
+description: This tool can be used to plot heatmap of gene expression data. The genes are chosen based on p-value, FDR, log FC and log CPM from edgeR output. 
+name: plotheatmap
+owner: earlhaminst
```
Can you please add the following line?
```yaml
remote_repository_url: https://github.com/TGAC/earlham-galaxytools/tree/master/tools/plotheatmap
```

nsoranzo commented on tools/plotheatmap/plotheatmap.xml:
```diff
@@ -0,0 +1,87 @@
+<tool id="plotheatmap" name="plotHeatmap" version="1.0">
+    <description>Plot heatmap</description>
+    <requirements>
```
The Travis test is failing because you're missing a requirement here:
```xml
        <requirement type="package" version="1.34.0">bioconductor-preprocesscore</requirement>
```

dzesikahoinkis committed:
Update plotheatmap tool

nsoranzo committed:
Change inputs format to tabular

nsoranzo commented:
Thanks @dzesikahoinkis!

nsoranzo commented:
Published on the Galaxy ToolShed: https://toolshed.g2.bx.psu.edu/view/earlhaminst/plotheatmap/
